demo/test.c

```diff
@@ -1221,10 +1222,11 @@ static int test_mp_montgomery_reduce(void)
             if (mp_cmp(&c, &d) != MP_EQ) {
 /* *INDENT-OFF* */
                printf("d = e mod a, c = e MOD a\n");
-               mp_to_decimal(&a, buf, sizeof(buf)); printf("a = %s\n", buf);
-               mp_to_decimal(&e, buf, sizeof(buf)); printf("e = %s\n", buf);
-               mp_to_decimal(&d, buf, sizeof(buf)); printf("d = %s\n", buf);
-               mp_to_decimal(&c, buf, sizeof(buf)); printf("c = %s\n", buf);
+               mp_to_decimal(&a, &buf); printf("a = %s\n", buf);
```

Comment:
all of those need to `free(buf)` before calling `mp_to_decimal()` again

Reply:
See above.